core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -381,7 +383,7 @@ class KafkaApis(val requestChannel: RequestChannel,
       }
 
       def produceResponseCallback(delayTimeMs: Int) {
-        if (produceRequest.acks == 0) {
+        if (acks == 0) {
```

Comment:
If we have to be careful about what we capture in this callback, it may be worth making it a top-level method.